src/Engine/ProtoAssociative/CodeGen.cs

```diff
@@ -6287,15 +6458,18 @@ private void EmitFunctionDefinitionNode(AssociativeNode node, ref ProtoCore.Type
             bool arrayIndexing = IsAssociativeArrayIndexing;
             IsAssociativeArrayIndexing = false;
 
-            // Handle static calls to reflect the original call
-            BuildRealDependencyForIdentList(graphNode);
 
-            if (node is FunctionDotCallNode)
+            if (subPass == ProtoCore.CompilerDefinitions.Associative.SubCompilePass.kNone)
             {
-                if ((node as FunctionDotCallNode).isLastSSAIdentListFactor)
+                // Handle static calls to reflect the original call
+                BuildRealDependencyForIdentList(graphNode);
+                if (node is FunctionDotCallNode)
```

Comment:
Instead of doing `is` and `as`, the other way is doing `as`:
```
var dotCallNode = node as FunctionDotCallNode;
if (dotCallNode != null)
{
    ...
}
```